File: backend/modules/sabor-familia/src/main/java/br/com/puc/saborfamilia/service/mensagem/dto/response/MensagemResponse.java
package br.com.puc.saborfamilia.service.mensagem.dto.response;

import br.com.puc.saborfamilia.service.perfil.dto.response.PerfilResumoResponse;
import java.time.LocalDateTime;

public record MensagemResponse(
  Long id,
  PerfilResumoResponse perfilRemetente,
  PerfilResumoResponse perfilDestinatario,
  String texto,
  LocalDateTime dataEnvio
) {
}

File: frontend/src/features/sabor_familia/dto/menssagem/response/MensagemResponse.ts
import type { PerfilResumoResponse } from "../../perfil/response/PerfilResumoResponse";

export interface MensagemResponse {
  id: number;
  perfilRemetente: PerfilResumoResponse;
  perfilDestinatario: PerfilResumoResponse;
  texto: string | null;
  dataEnvio: string;
  apagada: boolean;
}
